mtgjson5/price_builder.py

```diff
@@ -252,3 +297,159 @@ def build_prices(self) -> Tuple[Dict[str, Any], Dict[str, Any]]:
         local_zip_file.unlink()
 
         return archive_prices, today_prices
+
+    def build_prices_v2(
+        self,
+    ) -> Tuple[MtgjsonPricesV2Container, MtgjsonPricesV2Container]:
+        """
+        Build v2 format prices with proper archive workflow.
+        Downloads existing v2 archive, merges today's v2 data, prunes old records,
+        and uploads updated v2 archive.
+
+        Returns v2 price format:
+        - (all_prices_v2, today_prices_v2) as MtgjsonPricesV2Container objects
+
+        :return: Tuple of (archive_prices_v2, today_prices_v2)
+        """
+        LOGGER.info("Prices Build - Building Prices (V2 Format)")
+
+        # Get today's v2 price database
+        LOGGER.info("Building new v2 price data from providers")
+        today_prices_v2 = self.build_today_prices_v2()
+        if not today_prices_v2 or today_prices_v2.get_record_count() == 0:
+            LOGGER.warning("V2 pricing information failed to generate")
+            empty_v2 = MtgjsonPricesV2Container()
+            return empty_v2, empty_v2
+
+        if not MtgjsonConfig().has_section("Prices"):
+            return today_prices_v2, today_prices_v2
+
+        bucket_name = MtgjsonConfig().get("Prices", "bucket_name")
+        bucket_object_path_v2 = MtgjsonConfig().get(
+            "Prices", "bucket_object_path_v2", fallback="AllPricesv2.json.tar.xz"
+        )
+
+        # Download existing v2 archive (dict format: {provider: [records...]})
+        archive_dict = self.get_price_archive_data(bucket_name, bucket_object_path_v2)
```

Comment:
This is a full re implementation of the current archive process for pricing. I'd rather abstract that code and call it in the two places vs a copy.